spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java

```diff
@@ -292,6 +301,21 @@ public static SparkBigQueryConfig from(
     config.accessTokenProviderFQCN = getAnyOption(globalOptions, options, "gcpAccessTokenProvider");
     config.accessTokenProviderConfig =
         getAnyOption(globalOptions, options, "gcpAccessTokenProviderConfig");
+    try {
+      UserGroupInformation ugiCurrentUser = UserGroupInformation.getCurrentUser();
+      config.loggedInUserName =
+          com.google.common.base.Optional.of(ugiCurrentUser.getShortUserName());
+      config.loggedInUserGroups =
+          com.google.common.base.Optional.of(Sets.newHashSet(ugiCurrentUser.getGroupNames()));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```

Comment:
Please throw BigQueryConnectorException instead

Reply:
`BigQueryConnectorException` doesn't seem to allow wrapping another exception. The construction only accepts a String message. Should we maybe use something else?

Reply:
It has, see [this constructor](https://github.com/GoogleCloudDataproc/spark-bigquery-connector/blob/master/bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryConnectorException.java#L29). Notice you need to add a context relevant message (don't just use `e.getMessage()`) like "Failed to get current user"